### PR TITLE
Background subtraction

### PR DESCRIPTION
**Pull request by hroest**
Added a further method for background subtraction, based on what Skyline and EncyclopeDIA is apparently doing. See for example: https://www.biorxiv.org/content/early/2018/03/07/277822

```
We calculate trapezoidal
peak areas across Savitsky-Golay smoothed chromatograms. Analogous to Skyline,
peak intensities are background subtracted by removing a peak area rectangle with a
height equal to the largest intensity of either of the boundary edges. If the area of the
rectangle is larger than the area of the peak the intensity is set to zero.
```

@dmccloskey @pasqoo I would also like to change "vertical division" to "rectangular" as that is what we are really doing, a rectangular background subtraction. Any thoughts on this?

### CODE REVIEW

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h [hunk 1]
```diff
@@ -102,7 +118,8 @@ namespace OpenMS
     ///@}
 
     /** @name calculatePeakShapeMetrics() output
-      The calculatePeakShapeMetrics() method uses this struct to save its results.
+     *
```
Comment (pasqoo) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h]: Is the asterisk unwanted?

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h [hunk 2]
```diff
@@ -193,8 +210,9 @@ namespace OpenMS
     ///@}
 
     /** @name Constant expressions for parameters
-      Constants expressions used throughout the code and tests to set
-      the integration and baseline types.
+     *
```
Comment (pasqoo) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h]: Same here

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h [hunk 3]
```diff
@@ -205,8 +223,12 @@ namespace OpenMS
     static constexpr const char* INTEGRATION_TYPE_SIMPSON = "simpson";
     /// Baseline type: base to base
     static constexpr const char* BASELINE_TYPE_BASETOBASE = "base_to_base";
-    /// Baseline type: vertical division
+    /// Baseline type: vertical division (min of end points)
```
Comment (pasqoo) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h]: Maybe we should specify that this is kept for some backward compatibility, to not confuse future readers.

File: src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp [hunk 4]
```diff
@@ -249,19 +249,28 @@ namespace OpenMS
...
-          area += it->getPos();
+          pos_sum += it->getPos();
         }
         UInt n_points = std::distance(p.PosBegin(left), p.PosEnd(right));
-        area = (area - n_points * p.PosBegin(left)->getPos()) * delta_int / delta_pos + n_points * int_l;
+
+        // We construct the background area as the sum of a rectangular part
+        // and a triangle on top. The triangle is constructed as the sum of the
+        // line's y value at each sampled point: \sum_{i=0}^{n} (x_i - x_0)  * m
+        double rectangle_area = n_points * int_l;
```
Comment (pasqoo) [inline src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp]: This and the two following can be `const`.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h (hunk 3 above)
Comment (hroest) [reply src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h]: @pasqoo do you need this for backwards compatibility or can we just remove it?

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h [hunk 5]
```diff
@@ -205,8 +223,12 @@ namespace OpenMS
     static constexpr const char* INTEGRATION_TYPE_SIMPSON = "simpson";
     /// Baseline type: base to base
     static constexpr const char* BASELINE_TYPE_BASETOBASE = "base_to_base";
-    /// Baseline type: vertical division
+    /// Baseline type: vertical division (min of end points; only for backwards compatibility)
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h]: For naming, I would just suggest using "vertical_division" and "vertical_division_max" and not using "vertical_division_min".  The name "vertical_division" is pretty ubiquitous in most peak integration software (at least for small molecules).

Comment (hroest) [reply src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h]: I am ok with both, its just that using min/max would be less confusing for people.

Comment (dmccloskey) [reply src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h]: That works me.  Ready to be merged?

**Pull request merged**